url3472127e1f_SergeyKrivohatskiy_tiny_database_tgz-pJ8-ru_spbau_tinydb_bufferManager_IndexTestJ8/src/ru/spbau/tinydb/utils/Utils.java

```diff
@@ -0,0 +1,149 @@
+package ru.spbau.tinydb.utils;
+
+import ru.spbau.tinydb.btree.BxTreeEntry;
+import ru.spbau.tinydb.queries.Attribute;
+import ru.spbau.tinydb.queries.SecondLevelId;
+import ru.spbau.tinydb.table.Record;
+import ru.spbau.tinydb.table.Table;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * tiny_database
+ * Created by Sergey on 08.11.2014.
+ */
+public class Utils {
+    public static byte[] intToBytes(int i) {
+        return ByteBuffer.allocate(4).putInt(i).array();
+    }
+
+    public static int bytesToInt(byte[] intBytes) {
+        return ByteBuffer.wrap(intBytes, 0, 4).getInt();
+    }
+
+    public static byte[] stringToBytes(String value, int size) throws UnsupportedEncodingException {
+        byte[] res = new byte[size];
+        byte[] stringBytes = value.getBytes("US-ASCII");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;value&#39; (index: 0 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;size&#39;. The same parameter type is validated here for example: url3472127e1f_SergeyKrivohatskiy_tiny_database_tgz-pJ8-ru_spbau_tinydb_bufferManager_IndexTestJ8/src/ru/spbau/tinydb/common/DBANTLRErrorListener.java:28. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).